Sources/SwiftFormat/Rules/DontRepeatTypeInStaticProperties.swift

```diff
@@ -110,3 +67,38 @@ extension Finding.Message {
     "remove the suffix '\(type)' from the name of the variable '\(name)'"
   }
 }
+
+extension Syntax {
+  /// Returns the name of the immediately enclosing type of this node if there is one,
+  /// otherwise nil.
+  fileprivate func typeName() -> String? {
```

Comment:
This could be a computed property. Seems more swifty to me.

```suggestion
  fileprivate var typeName: String? {
```